libraries/custom_appbase/include/eosio/chain/exec_pri_queue.hpp

```diff
@@ -61,22 +76,56 @@ class exec_pri_queue : public boost::asio::execution_context
       should_exit_ = [](){ assert(false); return true; }; // should not be called when locking is disabled
    }
 
-   // called from appbase::application_base::exec poll_one() or run_one()
    template <typename Function>
-   void add(int priority, exec_queue q, size_t order, Function function) {
+   void add(int priority, exec_queue q, size_t order, Function&& function) {
       assert( num_read_threads_ > 0 || q != exec_queue::read_exclusive);
       prio_queue& que = priority_que(q);
-      std::unique_ptr<queued_handler_base> handler(new queued_handler<Function>(priority, order, std::move(function)));
+      std::unique_ptr<queued_handler_base> handler(new queued_handler<Function>(handler_id::unique, priority, order, std::forward<Function>(function)));
```

Comment:
I think it would be nicer to use the same pattern as the `add()` below, so the two functions are more similar and we don't have this `unique_ptr` stuff. 
```
      std::unique_lock g( mtx_, std::defer_lock );
      if (lock_enabled_ || q == exec_queue::read_exclusive) {
         // called directly from any thread for read_exclusive
         g.lock();
      }
      ...
      que.push( new queued_handler<Function>(...);
      }
```

Reply:
OK. I doubt it makes any significant difference though.